tests/integration_tests/functional/test_balloon.py

```diff
@@ -588,6 +588,7 @@ def _test_balloon_snapshot(context):
     # Now inflate the balloon with 20MB of pages.
     response = basevm.balloon.patch(amount_mib=20)
     assert basevm.api_session.is_status_no_content(response.status_code)
+    time.sleep(1)
```

Comment:
Maybe add a (short) comment here on why we sleep? The explanation in the commit is really good, I'm just not sure I will reach 

```
# sleep 1s to give MemoryMonitor thread time to update
```

or this is ugly but more explicit: `time.sleep(basevm._memory.MEMORY_SAMPLE_TIMEOUT_S)`

Reply:
I just thought about it and this may be even better: make a new method in the `Microvm` class, like so:

```python
def wait_for_memory_update(self):
    if self._memory_monitor:
        time.sleep(self._memory_monitor.MEMORY_SAMPLE_TIMEOUT_S)
```

(or maybe it can delegate to another method in `self._memory_monitor`, maybe that's cleaner but has the same effect).

and then in the balloon test:

```python
    basevm.wait_for_memory_update()
```